Packs/OPSWAT-Filescan/Integrations/OPSWATFilescan/OPSWATFilescan.py

```diff
@@ -0,0 +1,400 @@
+
+    if flow_id := args.get("flow_id"):
+        api_response = client.get_scan_result(flow_id)
+        successful_response = False
+
+        if api_response.get("allFinished", False):
+            successful_response = True
+
+        if successful_response:
+            if not is_valid_pass(api_response):
+                raise DemistoException("Invalid password!")
+
+            return PollResult(
+                response=build_reputation_result(api_response), continue_to_poll=False
+            )
+
+        return PollResult(
+            response=[
+                CommandResults(
+                    outputs=api_response,
+                    outputs_prefix=f"{INTEGRATION_CONTEXT_NAME}.Submission",
+                    outputs_key_field="flow_id",
+                    readable_output=f'Waiting for submission "{api_response.get("flow_id")}" to finish...',
+                ),
+            ],
+            continue_to_poll=True,
+            args_for_next_run={"flow_id": args.get("flow_id"), **args},
+        )
+    else:
+        return sample_submission(client, args)
+
+
+""" COMMANDS """
+
+
+def test_module_command(client: Client, *_) -> str:
+    """Performs a basic GET request to check if the API is reachable and authentication is successful."""
+    results = client.test_module()
+    if "accountId" in results:
+        return "ok"
+    raise DemistoException(f"\nTest module failed, {results}")
+
+
+def scan_command(client: Client, args: Dict[str, Any]):
+    return polling_submit_command(args=args, client=client)
+
+
+def search_query_command(client: Client, args: Dict[str, Any]):
+    def validate_args():
+        if page_size and (not page_size.isdigit() or int(page_size) not in [5, 10, 20]):
+            raise DemistoException("Page size value must be 5, 10 or 20")
+        if page and (not page.isdigit() or int(page) <= 0):
+            raise DemistoException("Page must be an integer and grater than 0")
+        if limit and (int(limit) <= 0 or int(limit) > 50):
+            raise DemistoException("Limit must be an integer and between 1 and 50")
+
+    items = []
+    query_string = args.get("query", "")
+    page_size = args.get("page_size")
+    page = args.get("page")
+    limit = args.get("limit", 10)
+
+    validate_args()
+
+    if page_size and not page:
+        page = 1
+        page_size = int(page_size)
+    elif not page_size and page:
+        page_size = 10
+        page = int(page)
+
+    if page_size and page:
+        items = client.get_search_query(query_string, page, page_size).get("items", [])
+    else:
+        page_size = 20
+        page = 1
+        limit = int(limit)
+        continue_query = True
+        while continue_query:
+            actual_items = client.get_search_query(query_string, page, page_size).get("items", [])
+            if len(actual_items) < 20 or len(items) >= limit:
+                continue_query = False
+            items += actual_items
```

Comment:
```suggestion
            items += actual_items
            page += 1
```

don't you need to raise the page each iteration?